### PR TITLE
Database backups

### LINKED ISSUE #1730: Backup databases before updating
We should hook into the `app:update` command so that a backup of the database is taken before we do anything.

At the very least this will cover ourselves and make some irreversible future migrations safer.

We could make use of https://github.com/ShawnMcCool/database-backup

### PR DESCRIPTION
**Pull request by jbrooksuk**
Closes #1730 

---

@GrahamCampbell the Laravel-Core package really needs the ability so that we can run a command **before** the rest of the command.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
https://github.com/GrahamCampbell/Laravel-Core/commit/20e9565a4bf35b045c1f2c52a689df326d160cec

**Comment by jbrooksuk** [comment]
Thanks @GrahamCampbell, this will make it much easier!

**Comment by jbrooksuk** [comment]
@GrahamCampbell this is now done.

**Comment by jbrooksuk** [comment]
Rebased.

@CachetHQ/owners can one of you merge this once the tests have completed please? I'm about to go and watch a film with the other half 👌

**Comment by GrahamCampbell** [comment]
Sure. Just got back in a few mins ago.

File: composer.json [hunk 1]
```diff
@@ -29,15 +29,16 @@
...
         "graham-campbell/markdown": "^6.1",
         "graham-campbell/exceptions": "^8.4",
         "guzzlehttp/guzzle": "^6.2",
         "jenssegers/date": "^3.1",
         "mccool/laravel-auto-presenter": "^4.2",
         "pragmarx/google2fa": "^0.7.1",
         "rcrowe/twigbridge": "^0.9.2",
-        "roumen/feed": "^2.10"
+        "roumen/feed": "^2.10",
+        "backup-manager/laravel": "^1.1"
```
Comment (GrahamCampbell) [inline composer.json]: not alphabetically sorted

**Pull request merged**